lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -52,9 +52,8 @@ def default_auth_type(settings)
         return AUTH_TYPES[:basic_auth] unless HammerCLIForeman::Sessions.enabled?
 
         url = settings.get(:_params, :host) || settings.get(:foreman, :host)
-        username = settings.get(:_params, :username) || settings.get(:foreman, :username)
         session = HammerCLIForeman::Sessions.get(url)
-        if !session.valid? && session.user_name == username && !session.auth_type.nil?
```

Comment:
I think we had that for this case:
 - I'm user `A`
 - I was logged in with `code grant auth`
 - I logged out
 - I want to log in again
 - I assume `code grant auth` again.

If we change that, then user `B` can be prompted for `code` even though he wanted `password grant auth`.

What do you think?

Reply:
When made this change I thought of this use case:

* I'm a user `A`
* In my config file, by default i have the setting - `username` set to `admin`.
* I login with user `A`, using the `code grant auth`
* Now, my session expires, and it checks `session.user_name == username`, where my `session.username` is `A` but my `username` is `admin`
* So it goes in the else part and takes `basic auth` 

Let me know your thoughts.

Reply:
Giving it a second thought, if in the case described, the user wants to relogin with a different auth source. If they pass in the command from hammer shell, then they will be redirected to the `foreman_api_reconnect` and correct auth type would be set. 

Let me know your thoughts. I think this needs a little brainstorming though :+1: